lib/screens/explore_screens/vouch/vouched_tab/interactor/usecases/get_user_citizenship_status_use_case.dart

```diff
@@ -0,0 +1,13 @@
+import 'package:seeds/datasource/local/settings_storage.dart';
+import 'package:seeds/datasource/remote/api/profile_repository.dart';
+import 'package:seeds/datasource/remote/model/profile_model.dart';
+import 'package:seeds/domain-shared/base_use_case.dart';
+
+class GetUserCitizenshipStatusUseCase extends NoInputUseCase<ProfileModel> {
+  final ProfileRepository _profileRepository = ProfileRepository();
+
+  @override
+  Future<Result<ProfileModel>> run() async {
+    return _profileRepository.getProfile(settingsStorage.accountName);
+  }
+}
```

Comment:
We already have this use case you could move it to shared use cases to avoid repeat code, also modify it to extends from `NoInput` use case base.

<img width="569" alt="image" src="https://user-images.githubusercontent.com/42857405/155070713-70443f36-f474-42a2-a7e9-9b9f3fe5e80a.png">

Reply:
Hi, change my mind about using this use case. Thank you